GoRogue/MapGeneration/Region.Static.cs

```diff
@@ -0,0 +1,120 @@
+        /// Creates a new Region from a GoRogue.Rectangle
+        /// </summary>
+        /// <param name="name">the name of this region</param>
+        /// <param name="rectangle">the rectangle containing this region</param>
+        /// <returns></returns>
+        public static Region FromRectangle(string name, Rectangle rectangle)
+        {
+            return Rectangle(name, rectangle.MinExtent, rectangle.Width, rectangle.Height);
+        }
+        public static Region RegularParallelogram(string name, Point origin, int width, int height, int rotationDegrees)
+        {
+            Point nw = origin;
+            Point ne = origin + new Point(width, 0);
+            Point se = origin + new Point(width * 2, height);
+            Point sw = origin + new Point(width, height);
+            Region area = new Region(name, se, ne, nw, sw);
+            area = area.Rotate(rotationDegrees, true, origin);
+            return area;
+        }
+
+        /// <summary>
+        /// Gets the points that are inside of the provided points
+        /// </summary>
+        /// <param name="outer">An IEnumerable of Points that form a closed region of any shape and size</param>
+        /// <returns>All points contained within the outer region</returns>
+        public static IEnumerable<Point> InnerFromOuterPoints(IEnumerable<Point> outer)
+        {
+            if(outer.Count() == 0)
+                return new List<Point>();
+            List<Point> points = new List<Point>();
+            outer = outer.OrderBy(x => x.X).ToList();
+            for (int i = outer.First().X + 1; i < outer.Last().X; i++)
+            {
+                List<Point> chunk = outer.Where(w => w.X == i).OrderBy(o => o.Y).ToList();
+                if(chunk.Count > 0)
+                {
+                    for (int j = chunk[0].Y; j <= chunk[^1].Y; j++)
+                    {
+                        points.Add(new Point(i, j));
+                    }
+                }
+            }
+
+            return points;
+        }
+
+
+
+        public static bool operator ==(Region left, Region right)
+        {
+            try
+            {
+                bool equals = left.Name == right.Name;
+                if (left.NorthWestCorner != right.NorthWestCorner) equals = false;
+                if (left.SouthWestCorner != right.SouthWestCorner) equals = false;
+                if (left.NorthEastCorner != right.NorthEastCorner) equals = false;
+                if (left.SouthEastCorner != right.SouthEastCorner) equals = false;
+                if (left.Center != right.Center) equals = false;
+                return equals;
+            }
+            catch (NullReferenceException)
+            {
+                return false;
+            }
+        }
+
+        public static bool operator !=(Region left, Region right)
+        {
+            try
+            {
+                return !(left == right);
+            }
+            catch (NullReferenceException)
```

Comment:
This is bad practice due to the performance implications of a try-catch, but works all the same.